dockstore-webservice/src/main/java/io/dockstore/webservice/core/Workflow.java

```diff
@@ -0,0 +1,130 @@
+/*
+ *    Copyright 2016 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package io.dockstore.webservice.core;
+
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorValue;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderBy;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * This describes one workflow in the dockstore, extending Entry with the fields necessary to describe workflows.
+ *
+ * Logically, this currently means one WDL or CWL document that may refer to tools in turn.
+ *
+ * @author dyuen
+ */
+@ApiModel(value = "Workflow", description = "This describes one workflow in the dockstore")
+@Entity
+@NamedQueries({
+        @NamedQuery(name = "io.dockstore.webservice.core.Workflow.findRegisteredById", query = "SELECT c FROM Workflow c WHERE c.id = :id AND c.isRegistered = true"),
+        @NamedQuery(name = "io.dockstore.webservice.core.Workflow.findAllRegistered", query = "SELECT c FROM Workflow c WHERE c.isRegistered = true"),
+        @NamedQuery(name = "io.dockstore.webservice.core.Workflow.findAll", query = "SELECT c FROM Workflow c"),
+        @NamedQuery(name = "io.dockstore.webservice.core.Workflow.searchPattern", query = "SELECT c FROM Workflow c WHERE ((c.defaultWorkflowPath LIKE :pattern) OR (c.description LIKE :pattern)) AND c.isRegistered = true") })
+@DiscriminatorValue("workflow")
+public class Workflow extends Entry {
+
+    @Column(nullable = false)
+    @ApiModelProperty(value = "This is the name of the workflow", required = true)
+    private String name;
+
+    // Add for new descriptor types
+    @Column(columnDefinition = "text")
+    @JsonProperty("workflow_path")
+    @ApiModelProperty(value = "This indicates for the associated git repository, the default path to the CWL document", required = true)
+    private String defaultWorkflowPath = "/Dockstore.cwl";
```

Comment:
For workflow, probably not. For tools, I think I'm comfortable with users registering them once and us looking for a CWL and WDL for the one entry. For workflows, I think we'll just allow them to have multiple entries (i.e. one repo with a workflow in two formats will just get two entries). I think that simplifies things.